src/interactions/slashCommands/mixins/CommandOptionWithChoices.ts

```diff
@@ -46,17 +51,43 @@ export abstract class ApplicationCommandOptionWithChoicesBase<T extends string |
 	 *
 	 * @param choices The choices to add
 	 */
-	public addChoices(choices: [name: string, value: T][]) {
+	public addChoices(choices: [name: string, value: T][]): Omit<this, 'setAutocomplete'> {
+		if (typeof this.autocomplete !== 'undefined') {
+			throw new RangeError('Autocomplete and choices are mutually exclusive to each other.');
+		}
+
 		ow(choices, `${ApplicationCommandOptionTypeNames[this.type]} choices`, choicesPredicate);
 
 		for (const [label, value] of choices) this.addChoice(label, value);
 		return this;
 	}
 
+	/**
+	 * Marks the option as autocompletable
+	 * @param autocomplete If this option should be autocompletable
+	 */
+	public setAutocomplete(autocomplete: boolean): Omit<this, 'addChoice' | 'addChoices'> {
```

Comment:
Why not add a default value of true here? This was done on the main lib and it feels like it makes sense for it to be that way

Reply:
Because this isn't the main library, and to be consistent with the rest of the builders

Reply:
Well then maybe this could be added to other builders? Not sure if it would make sense on all boolean options though

Reply:
Considering all of these booleans are optional by nature, if you call this method you explicitly want them, and making them a `toggleX` is not something I'm keen on. The benefits of having a default is what, not writing between 4-5 chars? What if the default changes on Discord's end? Then we have to update it too. Then users will complain that it's not behaving the same anymore. 🤷‍♂️

Reply:
The default would be set here in the library so Discord changing would not change anything at all. Every option will have autocomplete set to false by default and there's no doubt this is never going to change, so simply doing .setAutocomplete() seems logical to me